src/Compilers/Core/Portable/Collections/SmallDictionary.cs

```diff
@@ -724,7 +736,7 @@ IEnumerator IEnumerable.GetEnumerator()
         public struct Enumerator
         {
             private readonly Stack<AvlNode> _stack;
-            private Node _next;
+            private Node? _next;
             private Node _current;
 
             public Enumerator(SmallDictionary<K, V> dict)
```

Comment:
>Enumerator [](start = 19, length = 10)

Are each of these `struct Enumerator` types identical other than the `Current` property? If so, can we share the code easily?

Reply:
Potentially. But I'd like to keep this change focused on annotating existing behavior vs. refactoring. These changes are already big enough with just annotation. We can file a follow up issue if you like though.

---
In reply to: [356884067](https://github.com/dotnet/roslyn/pull/40268#discussion_r356884067) [](ancestors = 356884067)